api/src/main/kotlin/org/jetbrains/kotlinx/dl/dataset/embeddedDatasets.kt

```diff
@@ -318,6 +324,12 @@ private fun unzipDatasetPath(cacheDirectory: File, archive: File, dirRelativePat
     if (!dataDirectory.exists()) {
         Files.createDirectories(dataDirectory.toPath())
 
+        extractFromZipArchiveToFolder(archive.toPath(), toFolder)
```

Comment:
it looks like copy-paste and from the "else" block and could be refactored to the one function

Reply:
if I understand correctly the proposed logic the same piece of code will be executed in if and else branches and could be moved below after if-else condition

Reply:
But I also need to add that logic was to download and unzip the archive only if the data folder does not exist. Suppose the data folder exists no need to download it again. Yes, we have no special validation here for data: is it the correct dataset or maybe not.